### ISSUE TITLE
Failed to fetch due to CORS

### ISSUE THREAD

**Issue by Zachatoo** [jamiebrynes7/obsidian-todoist-plugin#191]
**Describe the bug**

When trying to use the plugin, you get an error that says

```
"TypeError: Failed to fetch."
```

When checking console, the error in there says

```
Access to fetch at 'https://api.todoist.com/rest/v1/tasks?filter=(today%20%7C%20overdue)' from origin 'app://obsidian.md' has been blocked by CORS policy:
Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.
If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
```

I believe this is due to this plugin using `fetch` instead of [request](https://github.com/obsidianmd/obsidian-api/blob/master/obsidian.d.ts#:~:text=export%20function%20request(request%3A%20RequestUrlParam%20%7C%20string)%3A%20Promise%3Cstring%3E%3B) or requestUrl from the Obsidian public api, which allows making requests without having to worry about CORS.

I recommend all (or most) instances of `fetch` should be replaced with `request` or `requestUrl` in order to resolve this issue.

**Additional context**

I'm creating this bug on behalf of someone else who is experiencing this who doesn't have a GitHub account and I don't use this plugin, so I cannot provide all of the information, but the bug should be pretty straightforward.

**Comment by moose-x86** [comment]
Hi,
I'm also experiencing this behavior.

**Comment by Zachatoo** [comment]
Some people have managed to resolve this with the following:

> Install the latest update. Todoist changed their security protocols. All Todoist connections stopped working and need to be updated.
> ... log out of and into Todoist again. You might need to do a web login before it will recognize it.

@moose-x86 May be worth trying the above.

**Comment by grepinsight** [comment]
I am also experiencing this

**Comment by grepinsight** [comment]
But clicking `Edit this block` and re-rendering fixes the issue!

**Comment by jamiebrynes7** [comment]
Hey all, I've changed the plugin to use `requestUrl` (released in `v1.10.1`), so hopefully that should resolve this issue.
